### PR TITLE
[fix](memory) Refactor memory allocated failure processing

### PR DESCRIPTION
**Pull request by Gabriel39**
## Proposed changes

In this PR,
1. Do not catch exception in operators.
2. Catch exception where we will get an exception.

### CODE REVIEW

**Comment by Gabriel39** [comment]
run buildall

File: be/src/vec/exec/scan/scanner_scheduler.cpp [hunk 1]
```diff
@@ -221,70 +221,85 @@ void ScannerScheduler::_scanner_scan(std::shared_ptr<ScannerContext> ctx,
...
     bool eos = false;
-    RuntimeState* state = ctx->state();
-    DCHECK(nullptr != state);
-    if (!scanner->is_init()) {
-        status = scanner->init();
-        if (!status.ok()) {
-            eos = true;
+    try {
+        doris::enable_thread_catch_bad_alloc++;
+        Defer defer {[&]() { doris::enable_thread_catch_bad_alloc--; }};
```
Comment (xinyiZzz) [inline be/src/vec/exec/scan/scanner_scheduler.cpp:226]: use a macro package `try catch`, easier to maintenance

File: be/src/runtime/fragment_mgr.cpp [hunk 2]
```diff
@@ -718,7 +718,21 @@ Status FragmentMgr::exec_plan_fragment(const TPipelineFragmentParams& params,
                                       this, std::placeholders::_1, std::placeholders::_2));
     {
         SCOPED_RAW_TIMER(&duration_ns);
-        auto prepare_st = context->prepare(params);
+        Status prepare_st = Status::OK();
+        try {
+            doris::enable_thread_catch_bad_alloc++;
+            Defer defer {[&]() { doris::enable_thread_catch_bad_alloc--; }};
```
Comment (xinyiZzz) [inline be/src/runtime/fragment_mgr.cpp:724]: use a macro package `try catch`, easier to maintenance

File: be/src/pipeline/task_scheduler.cpp [hunk 3]
```diff
@@ -128,6 +128,8 @@ void TaskScheduler::_do_work(size_t index) {
         auto status = Status::OK();
 
         try {
+            doris::enable_thread_catch_bad_alloc++;
+            Defer defer {[&]() { doris::enable_thread_catch_bad_alloc--; }};
```
Comment (xinyiZzz) [inline be/src/pipeline/task_scheduler.cpp:132]: use a macro package `try catch`, easier to maintenance

File: be/src/runtime/fragment_mgr.cpp (hunk 2 above)
Comment (xinyiZzz) [reply be/src/runtime/fragment_mgr.cpp:724]: Status st = RETURN_IF_CATCH_EXCEPTION(context->prepare(params));

**Comment by Gabriel39** [comment]
run buildall

**Review by xinyiZzz — APPROVED** [review]
LGTM

**Review by BiteTheDDDDt — APPROVED** [review]

**Pull request merged**